File: app/src/main/java/de/bitb/astroskop/model/converter/HouseConverter.java
package de.bitb.astroskop.model.converter;

import de.bitb.astroskop.model.enums.House;
import de.bitb.astroskop.model.enums.Zodiac;
import io.objectbox.converter.PropertyConverter;

public class HouseConverter implements PropertyConverter<House, Integer> {

    @Override
    public House convertToEntityProperty(Integer databaseValue) {
        for (House house : House.values()) {
            if (house.getId().equals(databaseValue)) {
                return house;
            }
        }
        return null;
    }

    @Override
    public Integer convertToDatabaseValue(House house) {
        return house.getId();
    }

}

File: app/src/main/java/de/bitb/astroskop/model/converter/HouseConverter.kt
package de.bitb.astroskop.model.converter

import de.bitb.astroskop.model.enums.House
import de.bitb.astroskop.model.enums.Zodiac
import io.objectbox.converter.PropertyConverter

class HouseConverter : PropertyConverter<House, Int> {

    override fun convertToEntityProperty(databaseValue: Int?): House? {
        for (house in House.values()) {
            if (house.id == databaseValue) {
                return house
            }
        }
        return null
    }

    override fun convertToDatabaseValue(house: House): Int? {
        return house.id
    }

}